test/utils.js

```diff
@@ -1,26 +1,84 @@
-                    if (result.result) result = result.result;
-                    if (typeof result === "number" && typeof expected === "number") {
-                        expect(result, `${formula} should equal ${expected}\n`).to.closeTo(expected, 0.00000001);
-                    } else {
-                        // For FormulaError
-                        if (expected.equals)
-                            expect(expected.equals(result), `${formula} should equal ${expected.error}\n`).to.equal(true);
-                        else
-                            expect(result, `${formula} should equal ${expected}\n`).to.equal(expected);
-                    }
-                })
+    generateTests: (parser, TestCase, type) => {
+        if(original){
+            const funs = Object.keys(TestCase);
+            funs.forEach(fun => {
+                it(fun, () => {
+                    const formulas = Object.keys(TestCase[fun]);
+                    formulas.forEach(formula => {
+                        const expected = TestCase[fun][formula];
+                        let result = parser.parse(formula, {row: 1, col: 1});
+                        if (result.result) result = result.result;
+                        if (typeof result === "number" && typeof expected === "number") {
+                            expect(result, `${formula} should equal ${expected}\n`).to.closeTo(expected, 0.00000001);
+                        } else {
+                            // For FormulaError
+                            if (expected.equals)
+                                expect(expected.equals(result), `${formula} should equal ${expected.error}\n`).to.equal(true);
+                            else
+                                expect(result, `${formula} should equal ${expected}\n`).to.equal(expected);
+                        }
+                    })
+                });
             });
+        }else {
+            const tests = allTests[type]
+            const funs = Object.keys(tests)
+            funs.forEach(fun => {
+                it(fun, () => {
+                    const formulas = Object.keys(tests[fun])
+                    formulas.forEach(formula => {
+                        let expected = tests[fun][formula]
+                        let result = parser.parse(formula, {row:1, col:1});
+                        if (result.result) result = result.result;
+                        //variable in nature, so I can not save them in a JSON
+                        if(formula === "HOUR(NOW())")
+                            expected = new Date().getHours();
+                        if(formula === "YEAR(NOW())")
+                            expected = new Date().getFullYear();
+                        
+                        if (typeof result === "number" && typeof expected === "number") {
+                            expect(result, `${formula} should equal ${expected}\n`).to.closeTo(expected, 0.00000001);
```

Comment:
Also print result

Reply:
They currently do, e.g:
 (AssertionError: SUM(A1:A1 A1:A1) should equal 0: expected 1 to be close to 0 +/- 1e-8)
would you like me to print out result a second time?

Reply:
ah k, ty